core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -145,40 +236,139 @@ public GeoBoundingBoxQueryBuilder ignoreMalformed(boolean ignoreMalformed) {
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (normalize == false) {
+            if (GeoUtils.isValidNumber(topLeft.getLat()) == false) {
+                validationException = addValidationError("Top part of top-left corner contains invalid number: " + topLeft.getLat(), validationException);
+            }
+            if (GeoUtils.isValidNumber(topLeft.getLon()) == false) {
+                validationException = addValidationError("Left part of top-left corner contains invalid number: " + topLeft.getLon(), validationException);
+            }
+            if (GeoUtils.isValidNumber(bottomRight.getLat()) == false) {
+                validationException = addValidationError("Bottom part of bottom-right corner contains invalid number: " + bottomRight.getLat(), validationException);
+            }
+            if (GeoUtils.isValidNumber(bottomRight.getLon()) == false) {
+                validationException = addValidationError("Top part of bottom-right corner contains invalid number: " + bottomRight.getLon(), validationException);
+            }
+        } else {
+            if (GeoUtils.isValidLatitude(topLeft.getLat()) == false) {
+                validationException = addValidationError("Top part of top-left corner contains invalid number: " + topLeft.getLat(), validationException);
+            }
+            if (GeoUtils.isValidLongitude(topLeft.getLon()) == false) {
+                validationException = addValidationError("Left part of top-left corner contains invalid number: " + topLeft.getLon(), validationException);
+            }
+            if (GeoUtils.isValidLatitude(bottomRight.getLat()) == false) {
+                validationException = addValidationError("Bottom part of bottom-right corner contains invalid number: " + bottomRight.getLat(), validationException);
+            }
+            if (GeoUtils.isValidLongitude(bottomRight.getLon()) == false) {
+                validationException = addValidationError("Top part of bottom-right corner contains invalid number: " + bottomRight.getLon(), validationException);
+            }
+        }
+
+        if (validationException == null) {
+            // all corners are valid after above checks - make sure they are in the right relation
+            if (topLeft.getLat() < bottomRight.getLat()) {
+                validationException = addValidationError("Top is below bottom corner: " +
+                        topLeft.getLat() + " vs. " + bottomRight.getLat(), validationException);
+            }
+        }
+
+        if (Strings.isEmpty(fieldName)) {
+            validationException = addValidationError("FieldName must be non-null and non-empty.", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    public Query doToQuery(QueryShardContext context) {
+        if (context.mapperService().smartNameFieldType(fieldName, context.getTypes()) == null){
```

Comment:
I believe I added this to make this query pass the tests in case there is no geo type defined in the index, should be in line with this: https://github.com/elastic/elasticsearch/issues/12016 (All IIRC and after a couple of minutes searching)